### ISSUE TITLE
Add moderator notes capability to Appeals

### ISSUE THREAD

**Issue by freiheit** [mastodon/mastodon#31235]
### Pitch

Add ability for moderators to add private (only visible to other moderators) notes to appeals. Similar to the notes on reports.

Simple example: multi-person moderation team and one moderator is discussing the appeal with the user (whether via mastodon DMs, email, or elsewhere), so they want to leave a note like "Leave this appeal alone. I'm discussing this with the user".

### Motivation

To help keep some kind of log of moderator activity within the Mastodon moderator interface somewhere, and to facilitate coordination between moderators on multi-moderator instances.

While a built-in mechanism for moderators to use the moderation interface for back-and-forth communication with the user about an appeal would be even better, I'm assuming that this is a much smaller and easier change to implement.

Cc: @This

**Comment by quintessence** [comment]
As an addendum for this: this would be an excellent use case for "Incident Report" / "Postmortem" style moderation reports, similar to what we see in the tech industry like:

https://postmortems.pagerduty.com/next_steps/

(Disclosure: I used to work for PagerDuty. That said, the doc is solid.)

There should be ways to flag a moderation report and/or appeal that is In Progress so that anyone on a distributed team knows that while they're catching up on all the chatter from when they were away. There should also be a way to create a higher level report with multiple sources (what if there are a bunch of reports that are about a single instance, and aren't user specific? etc.) so that aggregate trends can be found. (We currently do this manually and write reports as a private GitHub Wiki doc, but this is not ideal.)

**Comment by ThisIsMissEm** [comment]
I've actually been thinking about how to write up an "Incident Reports" feature, which is currently how several moderation teams I know of currently use report notes.

The idea here is that they write up a document that links together all the individual bits & pieces of reports relating to a single incident, linking everything together over time.

This deserves its own ticket but I haven't had a chance to research and write it up.

**Comment by quintessence** [comment]
What would you like to see in that ticket? I know I have thoughts on these as well (you know them) so I could probably do a write up.

**Comment by ThisIsMissEm** [comment]
The thing I'm mostly trying to figure out is how to make this buildable and not overtly complex.

**Comment by prabhaks** [comment]
Hi, are we still open to this feature? If yes, I would like to take a stab at it!

I think the ask is - moderators already have private notes on reports ([ReportNote](https://github.com/mastodon/mastodon/blob/main/app/models/report_note.rb)), but appeals have nothing equivalent, so we want to add it!

Approach is - I would mirror the existing report-notes pattern: a small `AppealNote` model + migration, an `Admin::AppealNotesController` (create/destroy), an `AppealNotePolicy` gated on manage_appeals (matching `AppealPolicy`), and the notes list + form rendered where an appeal is reviewed - since appeals have no show page, and an individual appeal is reviewed on the strike show page. 

Scope: I plan to keep v1 simple private mod-only notes, not the larger "Incident Reports" idea discussed earlier. 

Questions: 
- Does the v1 scope look okay?
- Does gating on `manage_appeals` sound right?
- Do we want to add create-and-approve/create-and-reject buttons or keep it to a plain add-note?
